frontend/src/components/GroupView/MiddleBar/GroupViewMiddleBarTaskQueue.tsx

```diff
@@ -15,8 +15,12 @@ const EmptyTaskQueue = (): React.ReactElement => (
   </div>
 );
 
-function renderTaskList(tasks: readonly Task[]): ReactNode {
-  return tasks.map((task) => <GroupTask key={task.id} original={task} memberName="" />);
+function renderTaskList(tasks: Task[]): ReactNode {
+  tasks.sort((a, b) => (a.order > b.order ? 1 : -1));
+
+  return tasks.map((task) => {
+    return <GroupTask key={task.id} original={task} memberName="" />;
+  });
```

Comment:
avoid `Task[]`, keep `readonly Task[]`. In the code below, do something like `[...tasks].sort((a, b) => (a.order > b.order ? 1 : -1)).map(...)`